activerecord/test/cases/base_test.rb

```diff
@@ -1131,11 +1131,14 @@ def test_benchmark_with_use_silence
   def test_clear_cache!
     # preheat cache
     c1 = Post.connection.schema_cache.columns("posts")
+    refute_equal 0, Post.connection.schema_cache.size
+
     ActiveRecord::Base.clear_cache!
+    assert_equal 0, Post.connection.schema_cache.size
+
     c2 = Post.connection.schema_cache.columns("posts")
-    c1.each_with_index do |v, i|
-      assert_not_same v, c2[i]
```

Comment:
The interning made this way of testing the schema was reloaded fail. 

I refactored the test, I believe it still offer the same confidence.